newlib/libc/machine/riscv/memmove-stub.c

```diff
@@ -1,14 +1,197 @@
-/* Copyright (c) 2019  SiFive Inc. All rights reserved.
+#if !defined(PREFER_SIZE_OVER_SPEED) && !defined(__OPTIMIZE_SIZE__)
+#include <string.h>
+#include <_ansi.h>
+#include <stddef.h>
+#include <limits.h>
+#include "xlenint.h"
+#include "sys/asm.h"
+#include "../../string/local.h"
 
-   This copyrighted material is made available to anyone wishing to use,
-   modify, copy, or redistribute it subject to the terms and conditions
-   of the FreeBSD License.   This program is distributed in the hope that
-   it will be useful, but WITHOUT ANY WARRANTY expressed or implied,
-   including the implied warranties of MERCHANTABILITY or FITNESS FOR
-   A PARTICULAR PURPOSE.  A copy of this license is available at
-   http://www.opensource.org/licenses.
-*/
 
-#if !defined(PREFER_SIZE_OVER_SPEED) && !defined(__OPTIMIZE_SIZE__)
-#include "../../string/memmove.c"
+static inline uint8_t __libc_aligned_xlen(void *dst, const void *src) {
```

Comment:
name is not correct, as this can return 1 when not xlen aligned. better to call it check_fast_alignment or so.